test/common/http/conn_manager_impl_test.cc

```diff
@@ -2149,9 +2149,8 @@ TEST_F(HttpConnectionManagerImplTest, ZeroByteDataFiltering) {
   Buffer::OwnedImpl zero;
   decoder->decodeData(zero, true);
 
-  // Continue.
-  EXPECT_CALL(*decoder_filters_[1], decodeData(_, true))
-      .WillOnce(Return(FilterDataStatus::StopIterationNoBuffer));
+  // The second filter decodeData() is not called with zero byte buffer.
```

Comment:
Without this change, the test case I added will fail in line 3196

Here is the detail;  add 3 decoder filters, 
1) filter[0]->decodeHeaders() returns STOP,  and is continued with ContinueDecoding,  
2) the filter[1]->decodeHeader() adds decodeData() and continue, 
3) filter[2]->decodeHeaders() is called, then filter[2]->decodeData() is called, 
4) filter[1]->decodeData() is called with zero byte buffer.

4) is not expected.  It causes crash in grpc_json filter in its  [assertion](https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc#L279) since end_stream is already called.

The reason step 4) is called it is because of this [line](https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L1472)

Since continueDecode() is called by the first filter,  the data is passed to the second filter again.

Reply:
Here is the calling graph:  "wrapper" refers to ConnectionManagerImpl::ActiveStreamFilter

filter0->decodeHeaders(),  stopped
   filter0->callback->continueDecode()
   wraper0->commonContinue
          wraper0->decodeHeaders()  {
                 filter1->decodeHeaders ()
                       addDecodeData()
                 filter2->decodeHeaders()
                 wrapper1->commonContinue() since it detects data is added by filter1  {
                      wrapper1->doData()
                          wrapper1->decodeData(filter1,..)
                                filter2->decodeData()
                }  // end of wrapper1->commonContinue
             }  // end of wrapper0->decodeHeaders
             wraper0->doData() is called
                   filter1->decodeData()      // this is wrong. it causes crash.